x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/search/TextExpansionQueryBuilder.java

```diff
@@ -188,14 +190,15 @@ protected QueryBuilder doRewrite(QueryRewriteContext queryRewriteContext) {
                         listener.onFailure(new IllegalStateException(warning.getWarning()));
                     } else {
                         listener.onFailure(
-                            new IllegalStateException(
+                            new ElasticsearchStatusException(
                                 "expected a result of type ["
                                     + TextExpansionResults.NAME
                                     + "] received ["
                                     + inferenceResponse.getInferenceResults().get(0).getWriteableName()
                                     + "]. Is ["
                                     + modelId
-                                    + "] a compatible model?"
+                                    + "] a compatible model?",
+                                RestStatus.BAD_REQUEST
```

Comment:
Let's use an `IllegalArgumentException` similar to what we do in SparseVectorQueryBuilder?